core/src/main/scala/com/evolutiongaming/kafka/flow/PartitionFlow.scala

```diff
@@ -206,17 +208,20 @@ object PartitionFlow {
       _ <- Log[F].debug("computing offset to commit")
 
       // find minimum offset if any
-      states <- cache.values
-
-      _ <- Log[F].debug(s"got ${states.size} states from cache")
-
-      stateOffsets <- states.values.toList.traverse { state =>
-        state flatMap (_.context.holding)
-      }
-
-      _ <- Log[F].debug(s"computed stateOffsets: $stateOffsets")
+      pickMin = CommutativeMonoid.instance[Option[Offset]](
+        none,
+        {
+          case (Some(x), Some(y)) => Some(x min y)
+          case (x @ Some(_), _)   => x
+          case (_, y)             => y
+        }
+      )
 
-      minimumOffset = stateOffsets.flatten.minimumOption
+      minimumOffset <- cache.foldMap {
+        case (_, Right(value)) => value.context.holding
+        case (key, Left(_)) =>
+          Async[F].raiseError(new Exception(s"trying to compute offset to commit but value for key $key is not ready"))
```

Comment:
I'm not sure this is safe to throw exceptions for `loading` elements

Reply:
We should never have `Left` case here. So either
1. this never happens (most likely)
2. this happens because we have a critical bug in our code. Then I'd like to find out about this bug ASAP instead of investigating for days why kafka-flow computes incorrect offset.

Reply:
Maybe log a warning or error here if there's a need to receive a signal about such behavior? Failing the application here could prove inconvenient later because at this point the framework has already persisted snapshots and, if failed here, would restart from the previous commit but with the updated state which can be potentially incorrect, more than committing a wrong offset (but moving further nonetheless)

Reply:
also you can just wait until it's loaded :)